src/sas/qtgui/Plotting/Plotter2D.py

```diff
@@ -396,6 +422,47 @@ def setSlicer(self, slicer, reset=True):
         # Clear current slicer
         if self.slicer is not None:
             self.slicer.clear()
+
+        # Clear the old slicer plots so they don't reappear later
+        if hasattr(self, '_item'):
+            item = self._item
+            if self._item.parent() is not None:
+                item = self._item.parent()
+
+            # Go through all items and see if they are a plot. The checks done here are not as thorough
+            # as GuiUtils.deleteRedundantPlots (which this takes a lot from). Will this cause problems?
+            # Primary concern is the check (plot_data.plot_role == DataRole.ROLE_DELETABLE) as I don't
+            # know what it does. The other checks seem to be related to keeping the new plots for that function
+            tempPlotsToRemove = []
+            slicer_type_id = 'Slicer' + self.data0.name
+            for itemIndex in range(item.rowCount()):
+                # GuiUtils.plotsFromModel tests if the data is of type Data1D or Data2D to determine
+                # if it is a plot, so let's try that
+                if isinstance(item.child(itemIndex).data(), (Data1D, Data2D)):
+                    # First take care of this item, then we'll take care of its children
+                    if hasattr(item.child(itemIndex).data(), 'type_id'):
+                        if slicer_type_id in item.child(itemIndex).data().type_id:
+                            # Store this plot to be removed later. Removing now
+                            # will cause the next plot to be skipped
+                            tempPlotsToRemove.append(item.child(itemIndex))
+                # It looks like the slicers are children of items that do not have data of instance Data1D or Data2D.
+                # Now do the children (1 level deep as is done in GuiUtils.plotsFromModel). Note that the slicers always
+                # seem to be the first entry (index2 == 0)
+                for itemIndex2 in range(item.child(itemIndex).rowCount()):
```

Comment:
What happens if the child has children? How the existing data explorer system works, that shouldn't be possible, but this limits our ability to search through larger tree structures.

Reply:
@jack-rooks, that seems like a good option.

Reply:
The most recent commit does this in a slightly different way since it doesn't seem "info" is actually a child of the data (checks to see if it has any text for use in the data explorer headings, if none of the children do then it deletes it).